docs/adr/0001-simple-multi-tenancy-using-a-single-opencloud-instance.md

```diff
@@ -0,0 +1,64 @@
+---
+title: "1. Simple Multi-Tenancy using a single OpenCloud Instance"
+---
+
+* Status: proposed
+* Deciders: [@micbar @butonic @dragotin @rhafer]
+* Date: 2025-05-20
+
+Technical Story: https://github.com/opencloud-eu/opencloud/issues/877
+
+## Context and Problem Statement
+
+To reduce resource usage and cost service providers want a single OpenCloud
+instance to host multiple tenants. Members of the same tenant should be able to
+only see each other when trying to share resources. A user can only be a member
+of a single tenant. Moving a user from one tenant to another is not supported.
+For the scope of this ADR we assume the tenants are overall rather small
+(sometimes just a single user, often less than 10). Also there is no need to
+support per tenant groups of users for now.
+
+## Decision Drivers
+
+* Low Resource Overhead: The solution should not require a lot of additional resources
+  (CPU, Memory, Storage) on the OpenCloud instance.
+* Implementation effort: The solution should be easy to implement and maintain.
+
+## Considered Options
+
+* A tenant id is assigned to every user. The tenant id is used to filter the
+  resources that are shown to the user. The tenant id will be a new part of the
+  CS3 UserId. The user <-> tenant id relation in maintained by the OIDC IDP and
+  the tenant id will be available as a claim in the OIDC tokens.
```

Comment:
Hm, do we want to rely on a claim issued by the IDP, or do we want to map the issuer to an internal tenant id, similar to how we shoul be doing it for the sub to internal user id mapping? I prefer the latter.